### DISCUSSION TITLE
About the calculation of the TMB?

### DISCUSSION THREAD

**Discussion by baike687** [PoisonAlien/maftools#662] [Q&A]
Excuse me, I'm a beginner in bioinformatics. 
I just wondering how the length of the human genome exon was acquired.
I see that in the script is 35.8Mb.

**Comment by PoisonAlien** [comment]
Hi,
It depends on the capture size used to target the coding regions. For example Agilent [sure select](https://www.agilent.com/en/product/next-generation-sequencing/hybridization-based-next-generation-sequencing-ngs/exome-probes/sureselect-human-all-exon-v7-232866) kit targets ~50MB of the coding regions (exons). Earlier versions used to target ~38MB of the coding regions.

**Reply by baike687** [reply]
Thanks a lot.
And should synonymous variants be taken into the calculation of TMB?

**Reply by PoisonAlien** [reply]
Usually only non-synonymous variants with functional impact are considered for TMB. That's what `maftools` does by default. 
But one can optionally include any kind of variants for TMB estimation (though not ideal).

**Reply by baike687** [reply]
Thanks, Anand. It's very helpful!

**Reply by PoisonAlien** [reply]
You are welcome and, all the very best.

**Reply by h170607** [reply]
Does this mean that maftools can automatically pick the appropriate mutation from the maf file to calculate the TMB without requiring us to process the maf file ourselves?
By the way, is there any suitable tool that can calculate the length of the target regions from the bed file?

**Comment by XiaoYan000** [comment]
Excuse me，
I wonder how to set the  captureSize for WGS data of mice？
Looking forward to your reply

**Reply by PoisonAlien** [reply]
Hi,
You could get the gtf file and sum up all the sizes of all exonic regions.
